lib/mayaUsd/ufe/UsdTRSUndoableCommandBase.h

```diff
@@ -46,22 +48,22 @@ namespace ufe {
 //   reparented.  In such a case, the prim in the command's scene item
 //   becomes stale, and the prim in the updated scene item should be used.
 //
-template<class V>
 class MAYAUSD_CORE_PUBLIC UsdTRSUndoableCommandBase : public Ufe::Observer,
-        public std::enable_shared_from_this<UsdTRSUndoableCommandBase<V> >
+        public std::enable_shared_from_this<UsdTRSUndoableCommandBase>
 {
 protected:
 
-    UsdTRSUndoableCommandBase(
-        const UsdSceneItem::Ptr& item, double x, double y, double z);
-    ~UsdTRSUndoableCommandBase() = default;
+    UsdTRSUndoableCommandBase(const UsdSceneItem::Ptr& item,
+        GfVec3f vec, UsdGeomXformOp::Type opType);
+
+    UsdTRSUndoableCommandBase(const UsdSceneItem::Ptr& item,
+        GfVec3d vec, UsdGeomXformOp::Type opType);
```

Comment:
Specialising funcs based on _what can be converted to GfVec3f/GfVec3d_, isn't the same as specialising based on the type imho. What is the motivation for avoiding const references here? (If there is a reason for it, you probably want to consider __vectorcall for the WIn32 build).